core/che-core-api-dto/src/main/java/org/eclipse/che/dto/server/DtoFactory.java

```diff
@@ -386,12 +388,37 @@ public Gson getGson() {
     private <T> DtoProvider<T> getDtoProvider(Class<T> dtoInterface) {
         DtoProvider<?> dtoProvider = dtoInterface2Providers.get(dtoInterface);
         if (dtoProvider == null) {
-            throw new IllegalArgumentException("Unknown DTO type " + dtoInterface);
+            if (!dtoInterface.isInterface()) {
+                throw new IllegalArgumentException(format("Only interfaces can be DTO, but %s is not", dtoInterface));
+            }
+
+            if (checkOnDtoAnnotation(dtoInterface)) {
+                throw new IllegalArgumentException(format("Provider of implementation for DTO type %s isn't found", dtoInterface));
+            } else {
+                throw new IllegalArgumentException(dtoInterface + " is not a DTO type");
+            }
         }
+
         return (DtoProvider<T>)dtoProvider;
     }
 
     /**
+     * Checks if dtoInterface or its parent have DTO annotation.
+     * @param dtoInterface
+     *          DTO interface
+     * @return true if onle dtoInterface or one of its parent have DTO annotation.
+     */
+    private boolean checkOnDtoAnnotation(Class dtoInterface) {
+        for (Class parent: dtoInterface.getInterfaces()) {
+            if (checkOnDtoAnnotation(parent)) {
+                return true;
+            }
+        }
+
+        return dtoInterface.isAnnotationPresent(DTO.class);
```

Comment:
I think it's better to check it before analyzing interfaces, for example:
```java
@DTO
interface A extends B, C { }
```

the code above will analyze B and C before analyzing A, which i think unreasonable.